pulp_container/app/registry_api.py

```diff
@@ -564,18 +564,95 @@ def create(self, request, path):
         """
         Create a new upload.
 
-        Note: We do not support monolithic upload.
         """
         _, repository = self.get_dr_push(request, path, create=True)
 
         if self.tries_to_mount_blob(request):
             response = self.mount_blob(request, path, repository)
+            return response
+        elif digest := request.query_params.get("digest"):
```

Comment:
I am quite puzzled by the specs. We can complete a monolithic upload in either a PUT or POST request. This is what you meant by the theoretical part, correct?

https://stackoverflow.com/a/57068286

Reply:
you not the only one ;)

monolithic upload can be done 2 ways:
1. Make a POST request with `digest` as request param and complete in one request the blob upload ( this is the theoretical part because i have no idea what clients use this, so basically this is implemented only based on docs without a way to prove this works fine). Also note that I am returning a 201 BlobResponse and not UploadResponse
2.  Make a POST request without `digest`  request param, then make a PUT request and send the chunk in the body.